src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -81,6 +85,13 @@ fun resolvePath(path: RsPath, lookup: ImplLookup = ImplLookup.relativeTo(path)):
         }
     }
 
+    val lifetimeArguments: List<Re>? =
+        path.typeArgumentList
+            ?.lifetimeList
+            ?.map { it.reference }
+            ?.map { it.resolve() as? RsLifetimeParameter }
+            ?.map { it?.let { ReEarlyBound.named(it) } ?: ReUnknown }
```

Comment:
Let's collapse all `map` calls into a single `map` call to avoid unnecessary creation of intermediate list objects

Reply:
Done it